src/ZoneServer/Skills/Handlers/Ardito/Taglio.cs

```diff
@@ -0,0 +1,179 @@
+	/// Handler for the Ardito skill Taglio.
+	/// </summary>
+	[SkillHandler(SkillId.Arditi_Taglio)]
+	public class Taglio : IDynamicCasted
+	{
+		private int _effectId;
+
+		/// <summary>
+		/// Handles skill casting - Start
+		/// </summary>
+		/// <param name="skill"></param>
+		/// <param name="caster"></param>
+		/// <param name="maxCastingTime"></param>
+		public void StartDynamicCast(Skill skill, ICombatEntity caster, float maxCastingTime)
+		{
+			if (!caster.TrySpendSp(skill))
+			{
+				caster.ServerMessage(Localization.Get("Not enough SP."));
+				return;
+			}
+
+			skill.IncreaseOverheat();
+			caster.SetAttackState(true);
+
+			var cancellationTokenSource = new CancellationTokenSource();
+
+			// Taglio: Tenacity
+			// Duration changed to 5 seconds
+			// Movement speed increase effect removed
+			// Cooldown increased by 10 seconds
+			if (!caster.Components.Get<AbilityComponent>().IsActive(AbilityId.Arditi19))
+			{
+				caster.Properties.Modify(PropertyName.MSPD_BM, 10);
+				Send.ZC_MSPD(caster);
+			}
+
+			// Why are we trusting the client?
+			// Find a way of getting the max Cast time for each skill...
+			cancellationTokenSource.CancelAfter(TimeSpan.FromSeconds(maxCastingTime));
+
+			this.AreaOfEffect(skill, caster, maxCastingTime, cancellationTokenSource.Token);
+		}
+
+		/// <summary>
+		/// Handles skill casting - Stop
+		/// </summary>
+		/// <param name="skill"></param>
+		/// <param name="caster"></param>
+		public void EndDynamicCast(Skill skill, ICombatEntity caster)
+		{
+			var buffComponent = caster.Components.Get<BuffComponent>();
+
+			if (buffComponent.Has(BuffId.Taglio_Buff))
+			{
+				buffComponent.Remove(BuffId.Taglio_Buff);
+			}
+
+			Send.ZC_NORMAL.UnkDynamicCastEnd(caster, skill.Id, 2.1f);
+			Send.ZC_STOP_SOUND(caster, "voice_war_atk_long_cast");
+			Send.ZC_NORMAL.GroundEffect_59(caster, caster.Direction, "Arditi_Taglio", skill.Id, caster.Position, this._effectId, false);
+
+			// Taglio: Tenacity
+			// Duration changed to 5 seconds
+			// Movement speed increase effect removed
+			// Cooldown increased by 10 seconds
+			if (!caster.Components.Get<AbilityComponent>().IsActive(AbilityId.Arditi19))
+			{
+				caster.Properties.Modify(PropertyName.MSPD_BM, -10);
+				Send.ZC_MSPD(caster);
+			}
+		}
+
+		/// <summary>
+		/// Executes the actual attack after a delay.
+		/// </summary>
+		/// <param name="skill"></param>
+		/// <param name="caster"></param>
+		/// <param name="maxCastingTime"></param>
+		/// <param name="cancellationToken"></param>
+		private async void AreaOfEffect(Skill skill, ICombatEntity caster, float maxCastingTime, CancellationToken cancellationToken)
+		{
+			await Task.Delay(150);
+
+			var buffComponent = caster.Components.Get<BuffComponent>();
+
+			if (buffComponent != null)
+			{
+				if (!buffComponent.Has(BuffId.Taglio_Buff))
+				{
+					buffComponent.Start(BuffId.Taglio_Buff, TimeSpan.FromSeconds(maxCastingTime));
+				}
+			}
+
+			this._effectId = ForceId.GetNew();
```

Comment:
You can't save skill values in instance fields, because the skill handlers are globally used classes. They aren't specifically created for each character, so you would be utilizing this exact variable for every single time this skill is casted by any entity, which could also happen multiple times at once.